src/Model/GServer.php

```diff
@@ -0,0 +1,1148 @@
+	 * @param string  $network Network value that is used, when detection failed
+	 *
+	 * @return boolean 'true' if server could be detected
+	 */
+	public static function detect(string $url, string $network = '')
+	{
+		$serverdata = [];
+
+		// When a nodeinfo is present, we don't need to dig further
+		$xrd_timeout = Config::get('system', 'xrd_timeout');
+		$curlResult = Network::curl($url . '/.well-known/nodeinfo', false, ['timeout' => $xrd_timeout]);
+		if ($curlResult->isTimeout()) {
+			DBA::update('gserver', ['last_failure' => DateTimeFormat::utcNow()], ['nurl' => Strings::normaliseLink($url)]);
+			return false;
+		}
+
+		$nodeinfo = self::fetchNodeinfo($url, $curlResult);
+
+		// When nodeinfo isn't present, we use the older 'statistics.json' endpoint
+		if (empty($nodeinfo)) {
+			$nodeinfo = self::fetchStatistics($url);
+		}
+
+		// If that didn't work out well, we use some protocol specific endpoints
+		if (empty($nodeinfo) || empty($nodeinfo['network']) || ($nodeinfo['network'] == Protocol::DFRN)) {
+			// Fetch the landing page, possibly it reveals some data
+			$curlResult = Network::curl($url, false, ['timeout' => $xrd_timeout]);
+			if ($curlResult->isSuccess()) {
+				$serverdata = self::analyseRootHeader($curlResult, $serverdata);
+				$serverdata = self::analyseRootBody($curlResult, $serverdata);
+			}
+
+			if (!$curlResult->isSuccess() || empty($curlResult->getBody())) {
+				DBA::update('gserver', ['last_failure' => DateTimeFormat::utcNow()], ['nurl' => Strings::normaliseLink($url)]);
+				return false;
+			}
+
+			if (empty($serverdata['network']) || ($serverdata['network'] == Protocol::DFRN)) {
+				$serverdata = self::detectFriendica($url, $serverdata);
+			}
+
+			if (empty($serverdata['network']) || ($serverdata['network'] == Protocol::ACTIVITYPUB)) {
+				$serverdata = self::detectMastodonAlikes($url, $serverdata);
+			}
+
+			// the 'siteinfo.json' is some specific endpoint of Hubzilla and Red
+			if (empty($serverdata['network']) || ($serverdata['network'] == Protocol::ZOT)) {
+				$serverdata = self::fetchSiteinfo($url, $serverdata);
+			}
+
+			// The 'siteinfo.json' doesn't seem to be present on older Hubzilla installations
+			if (empty($serverdata['network'])) {
+				$serverdata = self::detectHubzilla($url, $serverdata);
+			}
+
+			if (empty($serverdata['network'])) {
+				$serverdata = self::detectNextcloud($url, $serverdata);
+			}
+
+			if (empty($serverdata['network'])) {
+				$serverdata = self::detectGNUSocial($url, $serverdata);
+			}
+		} else {
+			$serverdata = $nodeinfo;
+		}
+
+		$serverdata = self::checkPoCo($url, $serverdata);
+
+		// We can't detect the network type. Possibly it is some system that we don't know yet
+		if (empty($serverdata['network'])) {
+			$serverdata['network'] = Protocol::PHANTOM;
+		}
+
+		// When we hadn't been able to detect the network type, we use the hint from the parameter
+		if (($serverdata['network'] == Protocol::PHANTOM) && !empty($network)) {
```

Comment:
Inside parentheses are unnecessary.

Reply:
But they increase readability.

Reply:
We don't have the same personal definition of "readability". 😅

Reply:
Obviously 😀